functions/src/modules/TE.functions.ts

```diff
@@ -67,19 +74,166 @@ export const processAllotment = functions.https.onCall(
     const coordinator = functions.config().coordinator;
 
     // Notify the assignee
-    await admin.database().ref(`/email/notifications`).push({
-      template: 'track-editing-allotment',
-      to: assignee.emailAddress,
-      bcc: coordinator.email_address,
-      params: {
-        tasks: tasksForEmail,
-        assignee: assignee,
-        comment: comment,
-        date: moment()
-          .utcOffset(coordinator.utc_offset)
-          .format('DD.MM'),
-        uploadURL: `${functions.config().website.base_url}/te/upload/`,
-      },
-    });
+    await admin
+      .database()
+      .ref(`/email/notifications`)
+      .push({
+        template: 'track-editing-allotment',
+        to: assignee.emailAddress,
+        bcc: coordinator.email_address,
+        params: {
+          tasks: tasksForEmail,
+          assignee: assignee,
+          comment: comment,
+          date: moment()
+            .utcOffset(coordinator.utc_offset)
+            .format('DD.MM'),
+          uploadURL: `${functions.config().website.base_url}/te/upload/`,
+        },
+      });
   }
 );
+
+const teUploadsBucket = `https://te.uploads.${
+  functions.config().storage['root-domain']
+}`;
+
+const editedBucket = `https://edited.${
+  functions.config().storage['root-domain']
+}`;
+
+export const handleTeSubmission = functions.storage
+  .bucket(teUploadsBucket)
+  .object()
+  .onFinalize(async object => {
+    const filePath = object.name;
+
+    const directories = path.dirname(filePath).split('/');
+    if (directories.length !== 2) {
+      throw new Error(
+        `File "${filePath}" is not uploaded to the appropriate folder`
+      );
+    }
+
+    const teUser = admin.auth().getUser(directories[0]);
+
+    teUser
+      .then(x => x.disabled)
+      .then(isActiveUser => {
+        if (!isActiveUser) {
+          throw new Error(`Invalid User`);
+        }
+      });
+
+    const emailAddress = await teUser.then(x => x.email);
+
+    const fileName = path.basename(filePath);
+
+    const list = helpers.extractListFromFilename(fileName);
+
+    const taskRef = admin
+      .database()
+      .ref(`/edited/${list}/${fileName}/trackEditing`);
+
+    //update task
+    await taskRef.update({
+      uploadPath: filePath,
+      status: 'Submitted',
+      submissionTimestamp: moment().format(),
+    });
+
+    // Getting the tasks list to be used when notifying the user
+    const task = (await taskRef.once('value')).val();
+
+    // Notify the user
+    await admin
+      .database()
+      .ref(`/email/notifications`)
+      .push({
+        template: 'te-submission-email',
+        to: emailAddress,
+        params: {
+          task,
+        },
+      });
+  });
+
+export const handleFeedbackUpdate = functions.database
+  .ref('/edited/{list}/{taskId}/trackEditing')
```

Comment:
Why not to trigger on `…/feedback` changes directly instead of testing `originalValues.feedback === changedValues.feedback` inside?